Automattic/wp-calypso
# Server Settings: Add separate buttons to clear edge cache and object cache

Fixes https://github.com/Automattic/dotcom-forge/issues/9078

## Proposed Changes

Per pcYYuD-1Sy-p2, we are introducing separate buttons on the Server Settings page for clearing the edge cache and the object cache. This PR also changes the heading of the "Cache" card to "Performance optimization".

This PR also converts `CacheCard` and the associated tests to Typescript, which makes for a slightly larger diff.

## Testing Instructions
1. Have an Atomic site
2. Navigate to `/hosting-config/:site_slug`
3. Assert that you see the following in the "Performance optimization" card.

<img width="516" alt="Screenshot 2024-09-18 at 12 48 35" src="https://github.com/user-attachments/assets/a230a3b7-c507-4f96-8dff-4ed606f70a1f">

4. Click `Clear all caches`
5. Assert that both that button and `Clear object cache` are disabled
6. Click `Clear edge cache`
7. Assert that it works as expected
